app/src/main/java/org/fossasia/openevent/general/event/EventDetailsViewModel.kt

```diff
@@ -35,6 +42,26 @@ class EventDetailsViewModel(private val eventService: EventService) : ViewModel(
                 }))
     }
 
+    fun loadMap(event: Event){
+        //location handling
+        val mapUrlInitial = "https://maps.googleapis.com/maps/api/staticmap?center="
+        val mapUrlProperties = "&zoom=12&size=1200x390&markers=color:red%7C"
+        val mapUrlMapType = "&markers=size:mid&maptype=roadmap"
+
+        val latLong: String = "" +event.latitude + "," + event.longitude
+
+        mapUrl.value = mapUrlInitial + latLong + mapUrlProperties + latLong + mapUrlMapType
+    }
+
+    fun loadMapIntent(event: Event, context: Context?){
+        val gmmIntentUri = Uri.parse("geo:<"+event.latitude+">,<"+event.longitude+">?q=<"+event.latitude+">,<"+event.longitude+">")
+        val mapIntent = Intent(Intent.ACTION_VIEW, gmmIntentUri)
+        mapIntent.`package` = "com.google.android.apps.maps"
+        if (mapIntent.resolveActivity(context?.packageManager) != null) {
+            mapIntentData.value = mapIntent
+        }
```

Comment:
Don't use Android classes in ViewModel

Reply:
Should I move the conditional to fragment?